subprojects/platform-play/src/main/java/org/gradle/play/internal/twirl/TwirlCompilerAdapterV13X.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.play.internal.twirl;
+
+import org.gradle.language.twirl.TwirlImports;
+import org.gradle.language.twirl.TwirlTemplateFormat;
+import org.gradle.scala.internal.reflect.ScalaCodecMapper;
+import org.gradle.scala.internal.reflect.ScalaMethod;
+import org.gradle.scala.internal.reflect.ScalaReflectionUtil;
+import org.gradle.util.CollectionUtils;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+class TwirlCompilerAdapterV13X extends TwirlCompilerAdapterV10X {
```

Comment:
This class could be a lot simpler if you called the Java Twirl API rather than the Scala API

Reply:
Thanks a lot! Do you mean https://github.com/playframework/twirl/pull/136 ? This is sure our next step plan.

Reply:
Yep!